src/screens/movieInfo/MovieInfo.tsx

```diff
@@ -0,0 +1,301 @@
+  const { favouriteMovies, setFavouriteMovies, watchlist, setWatchlist } =
+    useContext(ThemeAndStorageContext);
+
+  // memoized values
+  const isFavourite = useMemo(() => {
+    const isFavourite = favouriteMovies.some(
+      movieId => movieId === movieInfo.id + '',
+    );
+
+    return isFavourite;
+  }, [favouriteMovies]);
+
+  const inWatchlist = useMemo(() => {
+    const inWatchlist = watchlist.some(
+      movieId => movieId === movieInfo.id + '',
+    );
+
+    return inWatchlist;
+  }, [watchlist]);
+
+  // api handles
+  const loadSimilarMovies = useCallback(
+    async function loadSimilarMovies(page: number) {
+      const newSimilarMovies = await getSimiliarMovies(movieInfo.id + '', page);
+      setSimilarMovies([...similarMovies, ...newSimilarMovies]);
+    },
+    [similarMovies],
+  );
+
+  // onPress handlers
+  const addToFavouritesWrapper = useCallback(
+    (movieId: string) => {
+      addToFavourites(movieId);
+      setFavouriteMovies([...favouriteMovies, movieId]);
+    },
+    [favouriteMovies],
+  );
+
+  const removeFromFavouritesWrapper = useCallback(
+    (movieId: string) => {
+      setFavouriteMovies(favouriteMovies.filter(id => id !== movieId));
+
+      removeFromFavourites(movieId);
+    },
+    [favouriteMovies],
+  );
+
+  const addToWatchlistWrapper = useCallback(
+    (movieId: string) => {
+      addToWatchlist(movieId);
+
+      setWatchlist([...watchlist, movieId]);
+    },
+    [watchlist],
+  );
+
+  const removeFromWatchlistWrapper = useCallback(
+    (movieId: string) => {
+      setWatchlist(watchlist.filter(id => id !== movieId));
+
+      removeFromWatchlist(movieId);
+    },
+    [watchlist],
+  );
+
+  const _onFavouriteBtnPress = useMemo(() => {
+    return () =>
+      isFavourite
+        ? removeFromFavouritesWrapper(movieInfo.id + '')
+        : addToFavouritesWrapper(movieInfo.id + '');
+  }, [isFavourite]);
+
+  const _onWatchlistBtnPress = useMemo(() => {
+    return () =>
+      inWatchlist
+        ? removeFromFavouritesWrapper(movieInfo.id + '')
+        : addToFavouritesWrapper(movieInfo.id + '');
+  }, [inWatchlist]);
+
+  const _listRenderItem = useCallback(
+    ({ item }) => (
+      <MovieCard
+        movie={item}
+        onPress={() =>
+          navigation.push(ROUTES.MoviesStack.MovieInfo, {
+            movieId: item.id + '',
+          })
+        }
+      />
+    ),
+    [navigation],
+  );
+
+  // useEffects
+  useEffect(() => {
+    loadSimilarMovies(page);
+  }, [page]);
+
+  useEffect(() => {
+    (async () => {
+      setLoading(true);
+
+      const movieInfo = await getMovieInfo(params.movieId);
+      setMovieInfo(movieInfo);
+
+      const movieImages = await getMovieImages(params.movieId);
+      setMovieImages(movieImages);
+
+      const similarMovies = await getSimiliarMovies(params.movieId, 1);
+      setSimilarMovies(similarMovies);
+
+      setLoading(false);
+    })();
+  }, []);
+
+  if (loading) {
+    return <Loader animating={true} />;
```

Comment:
if using `animating` property why put this in condition?

Reply:
If `loading` is `true` then this component will only return the `Loader` else it will return what we actually want it to render

Reply:
My point is why not use the animating property directly, if you are using `animating` use it for the show-hide purpose directly

<img width="606" alt="Screenshot 2024-10-29 at 9 08 14 PM" src="https://github.com/user-attachments/assets/a5219c21-1cf1-423d-9c29-bf9d9906b79b">